core/src/toga/layout.py

```diff
@@ -0,0 +1,146 @@
+import json
+import toga
+import inspect
+from toga.style import Pack
+from xml.etree import ElementTree
+
+
+def parse_node_as_json(node):
+    return {
+        child.tag: parse_node_as_json(child) if len(child) else child.text or ""
+        for child in node
+    }
+
+
+def parse_attribute(app, element):
+    # If attribute is a list item and has children, parse it as dict
+    if element.tag == "item" and len(element):
+        return {element.tag: parse_node_as_json(element)}
+
+    # Get attributes
+    attributes = {**element.attrib}
+    attributes[element.tag] = []
+
+    # Get text inside node
+    if element.text.strip():
+        attributes[element.tag].append(element.text)
+
+    # For each child
+    for child in element:
+        # Parse child attribute
+        ele_attr = parse_attribute(app, child)[child.tag]
+        if type(ele_attr) != list:
+            ele_attr = [ele_attr]
+
+        # Save attribute
+        attributes[element.tag] = [*attributes[element.tag], *ele_attr]
+
+    return attributes
+
+
+def get_element_attrs(app, element):
+    # Object attributes
+    attributes = {**element.attrib}
+
+    # If there is text inside the tag, save it as attribute
+    if element.text and element.text.strip():
+        attributes["text"] = element.text
+
+    # Checking style and events props
+    for key, value in element.attrib.items():
+        # Handling events
+        if key.startswith("on_"):
+            # Handling inside class functions pointers
+            if value.startswith("."):
+                attributes[key] = getattr(app, value[1:])
+            continue
+
+        # Handling numeric values
+        if value.isnumeric():
+            attributes[key] = float(value)
+            continue
+
+        # Handling styles
+        if key == "style":
+            style = {}
+            for item in value.split(";"):
+                # Need the strip in case ";" is set at the end of style
+                if item.strip():
+                    a, v = item.split(":")
+                    style[a.strip()] = v.strip()
+            attributes["style"] = Pack(**style)
+            continue
+
+        # Handling app variables ref
+        if value.startswith("."):
+            attributes[key] = getattr(app, value[1:])
+            continue
+
+    return attributes
+
+
+def element_to_class(element_tag, attributes):
+    # Get TogaElement class
+    klass = getattr(toga, element_tag)
+
+    # Filter initial attributes
+    init_args = {
+        key: value
+        for key, value in attributes.items()
+        if key in inspect.getfullargspec(klass).args
+    }
```

Comment:
Not all attributes can be set in the constructor for all widgets, mostly those with content instead of children. So I got to extract those attributes that can be set on constructor to avoid runtime errors